src/test_report.py

```diff
@@ -27,6 +32,69 @@ def __init__(self, configs, args):
         api_token = os.getenv('API_TOKEN')
         self._db = kernelci.db.get_db(self._db_config, api_token)
         self._logger = Logger("config/logger.conf", "test_report")
+        self._email_host = args.email_host
+        self._email_port = int(args.email_port)
+        self._email_send_from = args.email_send_from
+        self._email_subject = args.email_subject
+        self._email_send_to = args.email_send_to
+
+    def create_email(self, email_content, send_to, send_from, email_subject):
+        email_msg = MIMEMultipart()
+        email_text = email.mime.text.MIMEText(email_content, "plain", "utf-8")
+        email_text.replace_header('Content-Transfer-Encoding', 'quopri')
+        email_text.set_payload(email_content, 'utf-8')
+        email_msg.attach(email_text)
+        if isinstance(send_to, list):
+            email_msg['To'] = ','.join(send_to)
+        else:
+            email_msg['To'] = send_to
+        email_msg['From'] = send_from
+        email_msg['Subject'] = email_subject
+        self._logger.log_message(logging.INFO, email_msg)
+        return email_msg
+
+    def smtp_connect(self):
+        email_user = os.getenv('EMAIL_USER')
+        email_password = os.getenv('EMAIL_PASSWORD')
+        try:
+            if self._email_port == 465:
+                email_server = smtplib.SMTP_SSL(self._email_host,
+                                                self._email_port)
+            else:
+                email_server = smtplib.SMTP(self._email_host,
+                                            self._email_port)
+                email_server.starttls()
+            email_server.login(email_user, email_password)
+        except (smtplib.SMTPAuthenticationError,
+                smtplib.SMTPConnectError,
+                smtplib.SMTPServerDisconnected) as error:
+            self._logger.log_message(logging.ERROR,
+                                     f"Connect or auth error: {error}.")
+        except (smtplib.SMTPHeloError,
+                smtplib.SMTPDataError,
+                smtplib.SMTPNotSupportedError,
+                RuntimeError) as error:
+            self._logger.log_message(logging.ERROR,
+                                     f"STMP server error: {error}.")
+        except (smtplib.SMTPException) as error:
+            self._logger.log_message(logging.ERROR,
+                                     f"Generic error connecting: {error}.")
+        else:
+            self._logger.log_message(logging.INFO, "Server Connected.")
+            return email_server
+
+    def send_mail(self, email_msg, email_server):
+        try:
+            email_server.send_message(email_msg)
+        except (smtplib.SMTPRecipientsRefused,
+                smtplib.SMTPSenderRefused) as error:
+            self._logger.log_message(logging.ERROR,
+                                     f"Error sending email: {error}.")
```

Comment:
If we want to have separate `exception` blocks for `SMTPRecipientsRefused` and `SMTPSenderRefused` we should take care of an appropriate error message.

Reply:
done